devops/setup-gpg/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: 'Setup GPG'
+description: 'Configures this action to run gpg with a given key and pass'
+inputs:
+  private-key:  # id of input
+    description: 'gpg private key'
+    required: true
+  key-pass:  # id of input
+    description: 'gpg key pass'
+    required: true
+  key-name:  # id of input
+    description: 'gpg key name'
+    required: true
+    default: "Aerospike"
+runs:
+  using: "composite"
+  steps:
+      - name: Set up GPG
+        env:
+          GPG_PRIVATE_KEY: ${{ inputs.private-key }}
+          GPG_KEY_PASS: ${{ inputs.key-pass }}
+          GPG_ID: ${{ inputs.key-name }}
+        run: |
+          # Setup gpg
+          mkdir -p ~/.gnupg
+          echo "$GPG_PRIVATE_KEY" | gpg --import --batch --yes
+          echo "$GPG_KEY_PASS"
+
+          # configure for non-interactive use
+          export GPG_TTY=no-tty
```

Comment:
If you define environment variables, they won't last after the step is finished because each step creates a new shell to run the commands. I'm not sure how this env variable is used, so it would help if you could clarify what it does

Reply:
Should we have .rpmmacro file instead of creating it.

Reply:
Julian, this is ok, these are just for the things that I am doing and is part of the reason I do it all in one step. I would have to repeat stuff a bunch. in the future I might break them out but for now I think it is fine.

Reply:
> Should we have .rpmmacro file instead of creating it.

we could. and probably should. at the point of writing this I did not know how to include files into composite workflows. I will add a ticket for this as it's not straight forward as there are some things to add to it anyhow.